platforms/netbsd/components.c

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2018 Kamil Rytarowski
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *   1. Redistributions of source code must retain the above copyright notice,
+ *      this list of conditions and the following disclaimer.
+ *
+ *   2. Redistributions in binary form must reproduce the above copyright
+ *      notice, this list of conditions and the following disclaimer in the
+ *      documentation and/or other materials provided with the distribution.
+ *
+ *   3. Neither the name of the copyright holder nor the names of its
+ *      contributors may be used to endorse or promote products derived from
+ *      this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#include <sys/param.h>
+#include <sys/types.h>
+#include <sys/device.h>
+#include <sys/kmem.h>
+
+#include "../../core/include/hax_core_interface.h"
+#include "../../core/include/config.h"
+
+extern struct cfdriver hax_vm_cd;
```

Comment:
Just curious: in which file is this variable actually declared? All declarations I can find are modified by `extern`.

Reply:
https://nxr.netbsd.org/xref/src/sys/sys/device.h#355

```
    375 #define	CFDRIVER_DECL(name, class, attrs)				\
    376 struct cfdriver __CONCAT(name,_cd) = {					\
    377 	.cd_name		= ___STRING(name),			\
    378 	.cd_class		= class,				\
    379 	.cd_attrs		= attrs,				\
    380 }
```

```
platforms/netbsd/hax_entry.c:CFDRIVER_DECL(hax_vm, DV_DULL, hax_vm_attrs);
platforms/netbsd/hax_entry.c:CFDRIVER_DECL(hax_vcpu, DV_DULL, hax_vcpu_attrs);
```